exonum-light-client/src/main/java/com/exonum/client/ExonumHttpClient.java

```diff
@@ -106,14 +115,91 @@ public String getUserAgentInfo() {
     });
   }
 
-  private static Request getRequest(HttpUrl url) {
+  @Override
+  public long getBlockchainHeight() {
+    BlocksResponse response = doGetBlocks(0, false, null, false);
+
+    return response.getBlocksRangeEnd();
+  }
+
+  @Override
+  public BlockResponse getBlockByHeight(long height) {
+    checkArgument(0 <= height, "Height can't be negative, but was {}", height);
+    HttpUrl url = urlBuilder()
+        .encodedPath(BLOCK)
+        .addQueryParameter("height", String.valueOf(height))
+        .build();
+    Request request = get(url);
+
+    return blockingExecuteAndParse(request, ExplorerApiHelper::parseGetBlockResponse);
```

Comment:
Will we get an exception as specified in the docs with the present code if height > maxHeight? Or shall we do additional checks?

Reply:
we will get `RuntimeException` as described in the docs. But we can handle this case explicitly.

Reply:
If it is a descriptive error saying why the operation has failed, I'd not bother.

Reply:
yes. It'll contain a message with "status: 404, body: Block for height: 12121212 not found "